File: gluecodium/src/test/resources/smoke/field_constructors/output/android/com/example/smoke/FieldCustomConstructorsMix.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
public final class FieldCustomConstructorsMix {
    @NonNull
    public String stringField;
    public int intField;
    public boolean boolField;
    public FieldCustomConstructorsMix(final int intValue, final double dummy) {
        FieldCustomConstructorsMix _other = createMe(intValue, dummy);
        this.stringField = _other.stringField;
        this.intField = _other.intField;
        this.boolField = _other.boolField;
    }
    public FieldCustomConstructorsMix(final int intField) {
        this.intField = intField;
        this.stringField = "nonsense";
        this.boolField = true;
    }
    private static native FieldCustomConstructorsMix createMe(final int intValue, final double dummy);
}

File: gluecodium/src/test/resources/smoke/field_constructors/output/android-kotlin/com/example/smoke/FieldCustomConstructorsMix.kt
/*

 *
 */

@file:JvmName("FieldCustomConstructorsMixExtensions")

package com.example.smoke


class FieldCustomConstructorsMix {
    @JvmField var stringField: String
    @JvmField var intField: Int
    @JvmField var boolField: Boolean





   constructor(intValue: Int, dummy: Double) {
        val _other = createMe(intValue, dummy)
        this.stringField = _other.stringField
        this.intField = _other.intField
        this.boolField = _other.boolField
    }



    constructor(intField: Int) {
        this.intField = intField
        this.stringField = "nonsense"
        this.boolField = true
    }





    companion object {

        @JvmStatic external fun createMe(intValue: Int, dummy: Double) : FieldCustomConstructorsMix
    }
}

